src/storage/src/source/mysql/replication/events.rs

```diff
@@ -233,7 +233,23 @@ pub(super) async fn handle_rows_event(
         let updates = [before_row.map(|r| (r, -1)), after_row.map(|r| (r, 1))];
         for (binlog_row, diff) in updates.into_iter().flatten() {
             let row = mysql_async::Row::try_from(binlog_row)?;
-            let packed_row = pack_mysql_row(&mut final_row, row, table_desc)?;
+            let packed_row = match pack_mysql_row(&mut final_row, row, table_desc) {
```

Comment:
This introduces a correctness issue. Imagine you have a table with time columns in MySQL and you do two transactions. Transaction `tx1` inserts an invalid row and `tx2` deletes it. These need to be translated into `(error, tx1_gtid, +1), (error, tx2_gtid, -1)` in Materialize, which cancel out! This means the source will be perfectly functional after you delete the problematic data.

In general whenever someone is inventing timestamps out of thin air or diffs out of thin air there is a very high chance definiteness is violated. In this particular case we are inventing a diff (because we hardcode it to `+1`) and also we drop some data on the floor because we early return. The correct way to handle these situations is something like the following, which will use the correct gtid AND diff regardless of whether the event is an error or not.


```suggestion
            let event = match pack_mysql_row(&mut final_row, row, table_desc) {
                Ok(row) => Ok(row),
                Err(err @ MySqlError::ValueDecodeError { .. }) => Err(DefiniteError::ValueDecodeError(err.to_string())),
                Err(err) => Err(err)?,
            };
            let data = (*output_index, event);
```